lib/BloqadeQMC/src/error.jl

```diff
@@ -36,18 +36,34 @@ jackknife(xs::Vector...) = jackknife(identity, xs...)
 
 
 function bootstrap(f::Function, xs::Vector...; nboot::Int=500)
+    
+    #check that all the vectors in xs are of equal length
+    num_vectors = length(xs)
+    vector_lengths = [length(xs[i]) for i in 1:length(xs)]
+    equal_length = all(vector_lengths[1] .== vector_lengths)
+    if equal_length == false
+        error("the vectors provided to 'bootstrap' are of different lengths")
+    end
```

Comment:
it can be done in one line

```suggestion
all(isequal(length(xs[1])), length.(xs)) || error("the vectors provided to 'bootstrap' are of different lengths")
```

Reply:
oh sick thanks, that's cool!